src/languages/go.js

```diff
@@ -45,14 +45,15 @@ function(hljs) {
       },
       {
         className: 'function',
-        beginKeywords: 'func', end: /\s*\{/, excludeEnd: true,
+        beginKeywords: 'func', end: /\B\b/,
         contains: [
           hljs.TITLE_MODE,
           {
             className: 'params',
             begin: /\(/, end: /\)/,
             keywords: GO_KEYWORDS,
-            illegal: /["']/
+            illegal: /["']/,
+            endsParent: true
```

Comment:
This will end the parent mode after you’d see a single set of parents. So that’s why it will never match a second set of parents.